### ISSUE TITLE
Missed OpConstantComposite support

### ISSUE THREAD

**Issue by iliya-diyachkov** [KhronosGroup/LLVM-SPIRV-Backend#29]
We have this issue at least in two OpenCL CTS tests: basic/progvar_func_scope, basic/progvar_prog_scope_uninit. llvm outputs this error: `LLVM ERROR: unable to write nop sequence of 36 bytes`.

It's because our writeNopData() implementation always returns false. A simple solution would be to change it to return true. However the real issue is the backend does not properly handle constant initializers of composites. It just puts the data in ELF flavor like this (from basic/progvar_func_scop):
```
...
        OpName %15 "test_bump.persistent"
...
        %14 = OpConstant %2 15
        %15 = OpVariable %10 CrossWorkgroup
...
        .p2align        6                               ; @test_bump.persistent
test_bump.persistent:
        .byte   97                              ; 0x61
        .zero   63
        .long   0                               ; 0x0
        .long   1                               ; 0x1
        .long   2                               ; 0x2
        .long   3                               ; 0x3
        .long   4                               ; 0x4
        .long   5                               ; 0x5
        .long   6                               ; 0x6
        .long   7                               ; 0x7
        .long   8                               ; 0x8
        .long   9                               ; 0x9
        .long   10                              ; 0xa
        .long   11                              ; 0xb
        .long   12                              ; 0xc
        .long   13                              ; 0xd
        .long   14                              ; 0xe
        .long   0                               ; 0x0
```
We need to implement OpConstantComposite support like in Translator:
```
...
               OpName %test_bump_persistent "test_bump.persistent"
...
   %uchar_97 = OpConstant %uchar 97
     %uint_0 = OpConstant %uint 0
     %uint_1 = OpConstant %uint 1
     %uint_2 = OpConstant %uint 2
     %uint_3 = OpConstant %uint 3
     %uint_4 = OpConstant %uint 4
     %uint_5 = OpConstant %uint 5
     %uint_6 = OpConstant %uint 6
     %uint_7 = OpConstant %uint 7
     %uint_8 = OpConstant %uint 8
     %uint_9 = OpConstant %uint 9
    %uint_10 = OpConstant %uint 10
    %uint_11 = OpConstant %uint 11
    %uint_12 = OpConstant %uint 12
    %uint_13 = OpConstant %uint 13
    %uint_14 = OpConstant %uint 14
   %uchar_98 = OpConstant %uchar 98
   %uint_100 = OpConstant %uint 100
    %v16uint = OpTypeVector %uint 16
%struct_mystruct_t = OpTypeStruct %uchar %v16uint
...
        %21 = OpConstantComposite %v16uint %uint_0 %uint_1 %uint_2 %uint_3 %uint_4 %uint_5 %uint_6 %uint_7 %uint_8 %uint_9 %uint_10 %uint_11 %uint_12 %uint_13 %uint_14 %uint_0
        %23 = OpConstantComposite %struct_mystruct_t %uchar_97 %21
%test_bump_persistent = OpVariable %_ptr_CrossWorkgroup_struct_mystruct_t CrossWorkgroup %23
```

**Comment by zuban32** [comment]
ConstantComposites should be initially covered by my 'removing custom IRTranslator' work

**Comment by iliya-diyachkov** [comment]
Other affected tests:
```
basic/progvar_prog_scope_misc
basic/kernel_memory_alignment_constant
basic/kernel_memory_alignment_local
basic/constant_source
basic/local_kernel_scope
basic/local_kernel_def
```

**Comment by iliya-diyachkov** [comment]
The issue was fixed mainly it #52. basic/progvar_func_scop passed after the patch for #74.
